src/core/memory.cc

```diff
@@ -407,14 +460,16 @@ class small_pool {
     unsigned _min_free;
     unsigned _max_free;
     unsigned _pages_in_use = 0;
+    bool _sampled_pool = false;
     page_list _span_list;
     static constexpr unsigned idx_frac_bits = 2;
 public:
-    explicit small_pool(unsigned object_size) noexcept;
+    explicit small_pool(unsigned object_size, bool is_sampled) noexcept;
     ~small_pool();
     void* allocate();
     void deallocate(void* object);
     unsigned object_size() const { return _object_size; }
+    bool is_sampled_pool() const { return _sampled_pool; }
```

Comment:
Please comment (even if the existing code sets a bad example with almost zero comments).

What is a "sampled pool"? (can reply by adding a good comment rather than the review)